server/src/main/java/org/candlepin/pinsetter/tasks/HypervisorUpdateJob.java

```diff
@@ -291,7 +291,9 @@ public void toExecute(JobExecutionContext context) throws JobExecutionException
 
                     if (factsUpdated || guestMigration.isMigrationPending() || typeUpdated) {
                         knownHost.setLastCheckin(new Date());
-                        guestMigration.migrate(false);
+                        if (guestMigration.isMigrationPending()) {
```

Comment:
This check is extraneous. We've already checked it in the parent conditional, and the method checks it before it executes.